### PR TITLE
Introduce the graveyard :headstone: 

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Adds back error messages for https://github.com/Lightning-AI/lightning/pull/14011 by introducing a "graveyard" module.

These would be removed in 2.0

Part of #15049 

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @borda

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Comment by awaelchli** [comment]
Blocked by #15071 but we are on it

File: src/pytorch_lightning/_graveyard/training_type.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+def _ttp_constructor(self: Any, *_: Any, **__: Any) -> None:
+    raise RuntimeError(
+        f"The `pl.plugins.{self._name}Plugin` class was removed in v1.8. Use `pl.strategies.{self._name}Strategy`"
+        " instead."
+    )
+
+
+def _patch_plugin_classes() -> None:
+    self = sys.modules[__name__]
+    for _name in (
```
Comment (carmocca) [inline src/pytorch_lightning/_graveyard/training_type.py:37]:
```suggestion
    for name in (
```

File: src/pytorch_lightning/_graveyard/training_type.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        "SingleDevice",
+        "SingleTPU",
+        "TPUSpawn",
+        "TrainingType",
+    ):
+        _plugin_name = _name + "Plugin"
+        _plugin_cls = type(_plugin_name, (object,), {"__init__": _ttp_constructor, "_name": _name})
+        setattr(self, _plugin_name, _plugin_cls)
+        # do not overwrite sys.modules as `pl.plugins` still exists. manually patch instead
+        setattr(pl.plugins, _plugin_name, _plugin_cls)
```
Comment (carmocca) [inline src/pytorch_lightning/_graveyard/training_type.py:58]:
```suggestion
        plugin_name = _name + "Plugin"
        plugin_cls = type(plugin_name, (object,), {"__init__": _ttp_constructor, "_name": name})
        setattr(self, plugin_name, plugin_cls)
        # do not overwrite sys.modules as `pl.plugins` still exists. manually patch instead
        setattr(pl.plugins, plugin_name, plugin_cls)
```

File: src/pytorch_lightning/_graveyard/training_type.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        "IPU",
+        "Parallel",
+        "DDPSharded",
+        "DDPSpawnSharded",
+        "SingleDevice",
+        "SingleTPU",
+        "TPUSpawn",
+        "TrainingType",
+    ):
+        plugin_name = _name + "Plugin"
```
Comment (carmocca) [inline src/pytorch_lightning/_graveyard/training_type.py:54]:
```suggestion
        plugin_name = name + "Plugin"
```

**Pull request merged**